lib/src/queue_list.dart

```diff
@@ -230,3 +248,17 @@ class QueueList<E> extends Object with ListMixin<E> implements Queue<E> {
     _head = 0;
   }
 }
+
+class _CastQueueList<S, T> extends QueueList<T> {
```

Comment:
This is creative. Tests, or it doesn't work. :)

Reply:
I was hoping to `extend ThingImAddingRetypeTo` as well 😄

Reply:
Done. Take a look - tests are basic.